source/Octopus.Tentacle.Tests.Integration/WorkspaceCleanerTests.cs

```diff
@@ -0,0 +1,183 @@
+                })
+                .Build(CancellationToken);
+
+            await Wait.For(() => !Directory.Exists(existingOrphanedWorkspaceDirectory), CancellationToken);
+        }
+
+        [Test]
+        [TentacleConfigurations(testDefaultTentacleRuntimeOnly: true)]
+        public async Task WhenTentacleStarts_WithWorkspaceYoungerThanThreshold_ThenWorkspaceIsLeftAlone(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            var cleanerDelay = TimeSpan.FromMilliseconds(500);
+            var deleteWorkspacesOlderThan = TimeSpan.FromMinutes(30);
+
+            var existingHomeDirectory = new TemporaryDirectory();
+            var existingOrphanedWorkspaceDirectory = GivenExistingOrphanedWorkspaceExists(existingHomeDirectory);
+            File.WriteAllText(ScriptWorkspace.GetLogFilePath(existingOrphanedWorkspaceDirectory), "Existing log file");
+
+            await using var clientAndTentacle = await tentacleConfigurationTestCase.CreateBuilder()
+                .WithTentacle(b =>
+                {
+                    b.WithHomeDirectory(existingHomeDirectory)
+                        .WithRunTentacleEnvironmentVariable(WorkspaceCleanerConfiguration.CleanerDelayEnvironmentVariableName, cleanerDelay)
+                        .WithRunTentacleEnvironmentVariable(WorkspaceCleanerConfiguration.DeleteWorkspacesOlderThanTimeSpanEnvironmentVariableName, deleteWorkspacesOlderThan);
+                })
+                .Build(CancellationToken);
+
+            await Task.Delay(1000, CancellationToken);
+
+            Directory.Exists(existingOrphanedWorkspaceDirectory).Should().BeTrue();
+        }
+
+        [Test]
+        [TentacleConfigurations(testDefaultTentacleRuntimeOnly: true)]
+        public async Task WhenTentacleIsRunning_WithWorkspaceThatBecomesOlderThanThreshold_ThenWorkspaceIsDeleted(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            var cleanerDelay = TimeSpan.FromMilliseconds(500);
+            var deleteWorkspacesOlderThan = TimeSpan.FromMilliseconds(500);
+
+            var existingHomeDirectory = new TemporaryDirectory();
+
+            await using var clientAndTentacle = await tentacleConfigurationTestCase.CreateBuilder()
+                .WithTentacle(b =>
+                {
+                    b.WithHomeDirectory(existingHomeDirectory)
+                        .WithRunTentacleEnvironmentVariable(WorkspaceCleanerConfiguration.CleanerDelayEnvironmentVariableName, cleanerDelay)
+                        .WithRunTentacleEnvironmentVariable(WorkspaceCleanerConfiguration.DeleteWorkspacesOlderThanTimeSpanEnvironmentVariableName, deleteWorkspacesOlderThan);
+                })
+                .Build(CancellationToken);
+
+            await Task.Delay(1000, CancellationToken);
+
+            var existingOrphanedWorkspaceDirectory = GivenExistingOrphanedWorkspaceExists(existingHomeDirectory);
```

Comment:
I think we need to revisit what we call orphaned as we should really be able to test this via script service and not by creating a workspace manually

Reply:
Indeed. Although, I am finding it hard to get this to work.

Now that we've fixed the "Is Running" issue, we can create an abandoned directory by killing a tentacle client.

That main test case is now covered in the renamed `WhenTentacleClientCrashesAndTentacleNeverGotToldToComplete_ThenWorkspaceShouldGetDeletedWhenScriptFinishesRunning` test.

But creating a directory like this would for the other tests has issues, as we would have to reuse the same tentacle (or create a new one that doesn't dispose until we finish).

I think this would be a good one to talk through when you're back.

Reply:
We reviewed this together via Zoom, and made the appropriate changes